### PR TITLE
DTSBPS-376: Spring boot upgrade

### PR DESCRIPTION
**Pull request by rishikrsharma**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBPS-376

### Change description ###
Springboot upgrade

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -181,13 +182,14 @@ dependencies {
...
   implementation group: 'org.springframework.boot', name: 'spring-boot-starter-validation'
-  implementation group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '3.1.2'
+  implementation group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '4.0.0'
 
   implementation group: 'io.springfox', name: 'springfox-swagger2', version: '3.0.0'
 
   implementation group: 'uk.gov.hmcts.reform', name: 'properties-volume-spring-boot-starter', version: '0.1.0', withoutSpringCloudContext
   implementation group: 'uk.gov.hmcts.reform', name: 'logging', version: versions.reformLogging
   implementation group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: versions.reformLogging
+  implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-bootstrap', version: '3.0.1'
```
Comment (mdayican) [inline build.gradle:192]: do we need this?

File: config/owasp/suppressions.xml [hunk 2]
```diff
@@ -31,5 +31,11 @@
     <cpe>cpe:/a:jetbrains:kotlin</cpe>
     <cve>CVE-2020-15824</cve>
   </suppress>
-
+  <suppress>
```
Comment (mdayican) [inline config/owasp/suppressions.xml:34]:
```suggestion
  <suppress until="2021-04-01">
```

File: build.gradle (hunk 1 above)
Comment (rishikrsharma) [reply build.gradle:192]: Yes

**Review by mdayican — APPROVED** [review]

File: config/owasp/suppressions.xml [hunk 3]
```diff
@@ -15,7 +15,7 @@
     <cve>CVE-2018-10237</cve>
   </suppress>
 
-  <suppress until="2021-03-01">
+  <suppress until="2021-05-01">
```
Comment (mdayican) [inline config/owasp/suppressions.xml:18]: you can remove this entry it is same with newly added one

**Review by mdayican — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (rishikrsharma) [reply build.gradle]: we don't need above lib. thanks.

File: config/owasp/suppressions.xml (hunk 3 above)
Comment (rishikrsharma) [reply config/owasp/suppressions.xml]: Remved.

**Pull request merged**